### DISCUSSION TITLE
Can I preview an app.db file?

### DISCUSSION THREAD

**Discussion by MiguelSOliveira** [simolus3/drift#2270] [Q&A]
If I have the path for an app.db file, can I preview the contents in the file easily?

I want to do a simple fetch on the contents of the file, so I can display to the user.

The use case is, I want the user to be able to restore their account to a backed up file, and I want to show them a preview of the contents of the file so they know what's in it before replacing the local one they are using.

**Comment by simolus3** [comment]
What exactly do you mean with "preview"? The app.db is a sqlite3 database file storing its contents in binary form. It is not really in a form that could be viewed directly.

To build a preview, you could open the file as a drift database, query the tables you expect to be there and then build the UI based on those results. What information to put in that preview probably depends on the exact information you're storing in your app, but in general you can just open the preview database just like you open the regular database file and then run queries to build a preview.
